### PR TITLE
Yandex Cloud support

### PR DESCRIPTION
**Pull request by knpsh**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

I've prepared some checks to support [Yandex Cloud ](https://cloud.yandex.com/en-ru/)platform Terraform [provider](https://registry.terraform.io/providers/yandex-cloud/yandex) in checkov.

### CODE REVIEW

**Comment by gruebel** [comment]
@knpsh thanks a lot for this great addition, I will review it during the week 🚀

**Comment by knpsh** [comment]
> @knpsh thanks a lot for this great addition, I will review it during the week 🚀

@gruebel Thanks! I'll be waiting for your feedback!

File: checkov/terraform/checks/resource/yandexcloud/ObjectStorageBucketPublicAccess.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        )
+
+    def scan_resource_conf(self, conf):
+        if 'acl' in conf.keys():
+            acl_block = conf['acl']
+            if acl_block in [["public-read"], ["public-read-write"]]:
+                return CheckResult.FAILED
+        if 'grant' in conf.keys():
+            grant_uri_block = conf['grant'][0]['uri']
+            if grant_uri_block in [["http://acs.amazonaws.com/groups/global/AllUsers"]]:
```
Comment (gruebel) [inline checkov/terraform/checks/resource/yandexcloud/ObjectStorageBucketPublicAccess.py:24]:
```suggestion
            if grant_uri_block == ["http://acs.amazonaws.com/groups/global/AllUsers"]:
```

File: checkov/terraform/checks/resource/yandexcloud/VPCSecurityGroupAllowAll.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+            name=name,
+            id=id,
+            categories=categories,
+            supported_resources=supported_resources,
+        )
+
+    def scan_resource_conf(self, conf):
+        if 'ingress' in conf.keys():
+            cidr_block = conf['ingress'][0]['v4_cidr_blocks']
+            if cidr_block[0][0] == "0.0.0.0/0":
```
Comment (gruebel) [inline checkov/terraform/checks/resource/yandexcloud/VPCSecurityGroupAllowAll.py:20]: I think you need to iterate over a possible list of IP ranges or is it not possible to have other IP ranges, when you have `0.0.0.0/0` inside?

File: checkov/terraform/checks/resource/yandexcloud/VPCSecurityGroupRuleAllowAll.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+            name=name,
+            id=id,
+            categories=categories,
+            supported_resources=supported_resources,
+        )
+
+    def scan_resource_conf(self, conf):
+        if conf['direction'][0] == 'ingress':
+            cidr_block = conf['v4_cidr_blocks']
+            if cidr_block[0][0] == "0.0.0.0/0":
```
Comment (gruebel) [inline checkov/terraform/checks/resource/yandexcloud/VPCSecurityGroupRuleAllowAll.py:20]: same here

File: tests/terraform/checks/resource/yandexcloud/example_ObjectStorageBucketEncryption/main.tf [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  bucket = "mybucket"
+
+}
+
+
+
+
+
+
+
```
Comment (gruebel) [inline tests/terraform/checks/resource/yandexcloud/example_ObjectStorageBucketEncryption/main.tf:26]: can you remove the empty lines here 🙂

**Review by gruebel — COMMENTED** [review]
nice, great work and thanks for putting in the time to add all the checks. I added a few comments, but all in all nice job 🚀

File: checkov/terraform/checks/resource/yandexcloud/IAMOrganizationElevatedMembers.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+            id=id,
+            categories=categories,
+            supported_resources=supported_resources,
+        )
+
+    def get_inspected_key(self) -> str:
+        return "role"
+
+    def get_forbidden_values(self) -> List[Any]:
+        return ["admin","editor","organization-manager.organizations.owner"]
```
[the remark this answers is not in the document]
Comment (knpsh) [reply checkov/terraform/checks/resource/yandexcloud/IAMOrganizationElevatedMembers.py]: Good eye, thanks!

File: checkov/terraform/checks/resource/yandexcloud/ObjectStorageBucketPublicAccess.py (hunk 1 above)
Comment (knpsh) [reply checkov/terraform/checks/resource/yandexcloud/ObjectStorageBucketPublicAccess.py]: Done

File: checkov/terraform/checks/resource/yandexcloud/VPCSecurityGroupAllowAll.py (hunk 2 above)
Comment (knpsh) [reply checkov/terraform/checks/resource/yandexcloud/VPCSecurityGroupAllowAll.py]: You are very much right, fixed this block.

File: checkov/terraform/checks/resource/yandexcloud/VPCSecurityGroupRuleAllowAll.py (hunk 3 above)
Comment (knpsh) [reply checkov/terraform/checks/resource/yandexcloud/VPCSecurityGroupRuleAllowAll.py]: Fixed it too.

File: tests/terraform/checks/resource/yandexcloud/example_ObjectStorageBucketEncryption/main.tf (hunk 4 above)
Comment (knpsh) [reply tests/terraform/checks/resource/yandexcloud/example_ObjectStorageBucketEncryption/main.tf]: Done!

**Review by gruebel — APPROVED** [review]
great job 🏆

**Comment by knpsh** [comment]
@gruebel Thanks a lot for support! 👍

**Pull request merged**